graphql_compiler/cost_estimation/statistics.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from abc import ABCMeta, abstractmethod
+
+import six
+
+
+@six.python_2_unicode_compatible
+@six.add_metaclass(ABCMeta)
+class Statistics(object):
+    """Abstract class for statistics regarding GraphQL objects and vertex field values.
```

Comment:
nit: GraphQL objects and their fields.

Reply:
Edges are GraphQL objects, but don't have any fields (as of now). I'll instead modify it to "GraphQL objects and vertex fields.`

Reply:
I think Pedro's point is that the type of a field can itself be a GraphQL object. So the statistics are covering objects and their fields, regardless of whether the objects in question represent vertices or edges or something else entirely.

Reply:
Oh now I see! But if a type of a field is a GraphQL object, that's either a scalar or an edge. We only have statistics if it's a scalar, so it's a lie to say we provide statistics about vertex fields referring to edges.